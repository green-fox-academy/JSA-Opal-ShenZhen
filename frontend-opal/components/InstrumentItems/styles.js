import { StyleSheet } from 'react-native';

export default StyleSheet.create({
  info: {
    display: 'flex',
    flexDirection: 'row',
    justifyContent: 'flex-end',
    paddingTop: 20,
  },
  greenfont: {
    color: 'green',
    flex: 1,
  },
  redfont: {
    color: 'red',
    flex: 1,
  },
  icon: {
    width: 14,
    height: 14,
  },
  text: {
    flex: 1,
  },
});
