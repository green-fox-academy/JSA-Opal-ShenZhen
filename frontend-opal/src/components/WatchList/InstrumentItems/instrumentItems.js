import React from 'react';
import { View, Text, Image } from 'react-native';
import PropTypes from 'prop-types';
import styles from './styles';

const infoItems = ({ infos }) => (
  <View>
    {infos.map(info => (
      <View key={info.name} style={styles.info}>
        <Text style={styles.text}>{info.name}</Text>
        <Text style={styles.text}>{info.num}</Text>
        <Text style={Number(info.increase) >= 0 ? styles.greenfont : styles.redfont}>
          {info.increase}%
        </Text>
        <Text style={styles.text}>{info.whole}</Text>
        <View style={styles.text}>
          <Image style={styles.icon} source={require('assets/transformation-icon.jpg')} />
        </View>
        <Image style={styles.icon} source={require('assets/icons-info.png')} />
      </View>
    ))}
  </View>
);

infoItems.propTypes = {
  infos: PropTypes.arrayOf(
    PropTypes.shape({
      name: PropTypes.string,
      num: PropTypes.string,
      increase: PropTypes.string,
      whole: PropTypes.string
    })
  ).isRequired
};

export default infoItems;
