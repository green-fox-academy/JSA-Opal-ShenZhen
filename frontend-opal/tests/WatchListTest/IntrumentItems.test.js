import React from 'react';
import renderer from 'react-test-renderer';

import InstrumentItems from '../../components/InstrumentItems/instrumentItems';

describe('<InstrumentItems />', () => {
  it('renders correctly', () => {
    const info = [{
      name: 'MSFT', num: '141.21', increase: '+0.23', whole: '1.2M',
    },
    {
      name: 'AMZN', num: '2108.2', increase: '-0.06', whole: '609K',
    },
    {
      name: 'AAPL', num: '250.7', increase: '+1.63', whole: '810K',
    }];
    const tree = renderer.create(<InstrumentItems infos={info} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
