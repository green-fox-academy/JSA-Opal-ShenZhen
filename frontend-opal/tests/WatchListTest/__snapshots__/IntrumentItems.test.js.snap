// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InstrumentItems /> renders correctly 1`] = `
<View>
  <View
    style={
      Object {
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "flex-end",
        "paddingTop": 20,
      }
    }
  >
    <Text
      style={
        Object {
          "flex": 1,
        }
      }
    >
      MSFT
    </Text>
    <Text
      style={
        Object {
          "flex": 1,
        }
      }
    >
      141.21
    </Text>
    <Text
      style={
        Object {
          "color": "green",
          "flex": 1,
        }
      }
    >
      +0.23%
    </Text>
    <Text
      style={
        Object {
          "flex": 1,
        }
      }
    >
      1.2M
    </Text>
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "height": 14,
            "width": 14,
          }
        }
      />
    </View>
    <Image
      source={1}
      style={
        Object {
          "height": 14,
          "width": 14,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "flex-end",
        "paddingTop": 20,
      }
    }
  >
    <Text
      style={
        Object {
          "flex": 1,
        }
      }
    >
      AMZN
    </Text>
    <Text
      style={
        Object {
          "flex": 1,
        }
      }
    >
      2108.2
    </Text>
    <Text
      style={
        Object {
          "color": "red",
          "flex": 1,
        }
      }
    >
      -0.06%
    </Text>
    <Text
      style={
        Object {
          "flex": 1,
        }
      }
    >
      609K
    </Text>
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "height": 14,
            "width": 14,
          }
        }
      />
    </View>
    <Image
      source={1}
      style={
        Object {
          "height": 14,
          "width": 14,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "flex-end",
        "paddingTop": 20,
      }
    }
  >
    <Text
      style={
        Object {
          "flex": 1,
        }
      }
    >
      AAPL
    </Text>
    <Text
      style={
        Object {
          "flex": 1,
        }
      }
    >
      250.7
    </Text>
    <Text
      style={
        Object {
          "color": "green",
          "flex": 1,
        }
      }
    >
      +1.63%
    </Text>
    <Text
      style={
        Object {
          "flex": 1,
        }
      }
    >
      810K
    </Text>
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "height": 14,
            "width": 14,
          }
        }
      />
    </View>
    <Image
      source={1}
      style={
        Object {
          "height": 14,
          "width": 14,
        }
      }
    />
  </View>
</View>
`;
